crates/wasmtime/src/engine.rs

```diff
@@ -403,6 +403,28 @@ impl Engine {
             };
         }
 
+        // There is no is_s390x_feature_detected macro yet, so for now
+        // we use getauxval from the libc crate directly.
+        #[cfg(all(target_arch = "s390x", target_os = "linux"))]
+        {
+            let v = unsafe { libc::getauxval(libc::AT_HWCAP) };
+            const HWCAP_S390X_VXRS_EXT2: libc::c_ulong = 32768;
+
+            enabled = match flag {
+                // There is no separate HWCAP bit for mie2, so assume
+                // that any machine with vxrs_ext2 also has mie2.
+                "has_vxrs_ext2" | "has_mie2" => Some((v & HWCAP_S390X_VXRS_EXT2) != 0),
+                // fall through to the very bottom to indicate that support is
+                // not enabled to test whether this feature is enabled on the
+                // host.
+                _ => None,
+            }
+        }
+        #[cfg(all(target_arch = "s390x", not(target_os = "linux")))]
```

Comment:
Instead of a maze of `#[cfg]` statements here, while you're at it could this be updated to use `cfg_if!`?

Either that or say above `let mut enabled = None` or something like that. Basically I don't think it's worth it to have lots of negated cfg's to handle this variable.

Reply:
I think I tried the `let mut enabled = None` approach, which caused the compiler to complain about a value that was written, but not read, hence the current approach.

Reply:
With a compiler warning I think it's fine to have a local `#[allow]` for something like this, that's easier to maintain than a complicated `#[cfg]` clause